src/app/api/boosts/claim/route.ts

```diff
@@ -0,0 +1,116 @@
+import { PrismaClient } from '@prisma/client';
+import { NextResponse, type NextRequest } from 'next/server';
+import '@/utils/helper';
+import { Network, Alchemy } from "alchemy-sdk";
+
+const settings = {
+    apiKey: process.env.ALCHEMY_ID,
+    network: Network.BASE_MAINNET,
+};
+
+const alchemy = new Alchemy(settings);
+
+async function ownsNFT(userAddress: string, contract: string) {
+    const response = await alchemy.nft.verifyNftOwnership(userAddress, contract);
+    return response;
+}
+
+async function hasToken(userAddress: string, contract: string) {
+    const contracts = [contract];
+
+    const tokenBalanceRes = await alchemy.core.getTokenBalances(userAddress, contracts);
+    const filterByBalance = tokenBalanceRes.tokenBalances.filter((balance: any) => Number(balance.tokenBalance) > 0);
+    return filterByBalance.length > 0;
+}
+
+const prisma = new PrismaClient();
```

Comment:
ah we are also doing this everywhere in our API. This should really be a singleton.

cc @amit-shah-cb